python/ray/tests/test_client_warnings.py

```diff
@@ -0,0 +1,65 @@
+from ray.util.client.ray_client_helpers import ray_start_client_server
+from ray.util.client.worker import TASK_WARNING_THRESHOLD
+from ray.util.debug import _logged
+
+import numpy as np
+import pytest
+
+import unittest
+
+
+@pytest.fixture(autouse=True)
+def reset_debug_logs():
+    """Resets internal state of ray.util.debug so that warning can be tested
+    for more than once in the same process"""
+    _logged.clear()
+
+
+class LoggerSuite(unittest.TestCase):
+    """Test client warnings are raised when many tasks are scheduled"""
+
+    def testManyTasksWarning(self):
+        with ray_start_client_server() as ray:
+
+            @ray.remote
+            def f():
+                return 42
+
+            with self.assertWarns(UserWarning) as cm:
+                for _ in range(TASK_WARNING_THRESHOLD + 1):
+                    f.remote()
+            assert f"More than {TASK_WARNING_THRESHOLD} remote tasks have " \
+                "been scheduled." in cm.warning.args[0]
+
+    def testNoWarning(self):
+        with ray_start_client_server() as ray:
+
+            @ray.remote
+            def f():
+                return 42
+
+            no_warning_raised = False
+            try:
+                with self.assertWarns(UserWarning) as cm:
+                    for _ in range(TASK_WARNING_THRESHOLD):
+                        f.remote()
+            except AssertionError:
+                # assertion above failed, meaning no warning was raised
+                no_warning_raised = True
+            if not no_warning_raised:
```

Comment:
Can you just raise the AssertionError in the Except block?

Reply:
Actually, just remove this entire thing out of the try Except, it should just be:
```
for _ in range(TASK_WARNING_THRESHOLD):
    f.remote()
```

Reply:
Tried that out and looks like the test won't fail by default even with the warn (although maybe the CI runs with an extra flag that would do so)

Adjusted it so that its a bit less messy looking and works without any extra flags

Reply:
I totally forgot that these are warnings! This makes sense!